packages/tldraw/src/lib/shapes/image/ImageShapeUtil.tsx

```diff
@@ -124,16 +107,17 @@ export class ImageShapeUtil extends BaseBoxShapeUtil<TLImageShape> {
 
 					ctx.drawImage(image, 0, 0)
 					setStaticFrameSrc(canvas.toDataURL())
-					setLoadedSrc(url)
+					setLoaded({ src: url, isPlaceholder })
+					if (url instanceof ReferenceCounterWithFixedTimeout) url.release()
```

Comment:
one of the reasons i'm a big skeptical about this reference counting stuff is i'm really not sure that the juice is worth the squeeze in terms of a slightly better experience for people who don't have any connectivity for minutes at a time (is it even a better experience? has their up/download actually just silently failed at that point?) vs the added api complexity and manual memory management footgun potential. 

just reading this back for example, i've noticed a memory leak here: if `cancelled` is true, we never call `.release()`.

Reply:
discussed offline: we'll remove this complexity and we're gonna just do a simple timeout to clear out memory